net/base_socket.h

```diff
@@ -60,8 +61,8 @@ struct sockaddr_storage {
     explicit sockaddr_storage(const sockaddr_in6& addr) {
         *((sockaddr_in6*) &store) = addr;
     }
-    explicit sockaddr_storage(const sockaddr& addr) {
-        *((sockaddr*) &store) = addr;
```

Comment:
`sockaddr` is always supposed to be used as a pointer. Otherwise might lose information.